src/Controller/ProxyCallbackController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_authentication\Controller;
+
+use Drupal\cas\Controller\ProxyCallbackController as CASProxyCallbackController;
+use Drupal\cas\Service\CasHelper;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class ProxyCallbackOEAuthenticationController.
+ */
+class ProxyCallbackController extends CASProxyCallbackController {
+
+  /**
+   * The configuration factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Database\Connection $database_connection
+   *   The database service.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request_stack
+   *   The Symfony request stack.
+   * @param \Drupal\cas\Service\CasHelper $cas_helper
+   *   The CasHelper.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The Config Factory.
+   */
+  public function __construct(Connection $database_connection, RequestStack $request_stack, CasHelper $cas_helper, ConfigFactoryInterface $configFactory) {
+    parent::__construct($database_connection, $request_stack, $cas_helper);
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database'),
+      $container->get('request_stack'),
+      $container->get('cas.helper'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * Route callback for the ProxyGrantingTicket information.
+   *
+   * This function stores the incoming PGTIOU and pgtId parameters so that
+   * the incoming response from the EULogin Server can be looked up.
+   */
+  public function callback() {
+    $config = $this->configFactory->get('oe_authentication.settings');
+    if ($config->get('protocol') === 'cas') {
```

Comment:
This sounds strange. Since we are the ones using a specific protocol, should we not check if $config->get('protocol') !== 'eulogin' ?

Reply:
Make totally sense.